fpdf/fpdf.py

```diff
@@ -2570,6 +2571,44 @@ def skew(self, ax=0, ay=0, x=None, y=None):
             )
             yield
 
+    @check_page
+    @contextmanager
+    def mirror(self, angle, origin=(None, None)):
+        """
+        This method allows to perform a reflection transformation over a given mirror line.
+        It must be used as a context-manager using `with`:
+
+            with mirror(angle="SOUTH", origin=(15,15)):
+                pdf.something()
+
+        The mirror transformation affects all elements which are printed inside the indented
+        context (with the exception of clickable areas).
+
+        Args:
+            angle: (fpdf.enums.Angle): the direction of the mirror line
+            origin (Sequence[float, float]): a point on the mirror line
+        """
+        angle = Angle.coerce(angle)
+        x, y = origin
+
+        if x is None:
+            x = self.x
+        if y is None:
+            y = self.y
```

Comment:
It's interesting that you thought about that usage.
I'm not sure many users will find it useful.

I'm more inclined in requiring `origin` to always be provided,
but I'm curious to know your thoughts.

Reply:
I hadn't fully understood the coordinate attributes so I agree, I don't think this is a common use-case